sdlib/d/gc/thread.d

```diff
@@ -181,30 +181,32 @@ public:
 		import d.gc.hooks;
 		__sd_gc_pre_stop_the_world_hook();
 
-		stopTheWorldMutex.lock();
+		// Prevent any new threads from being created
+		createThreadLock.exclusiveLock();
 
 		uint count;
-		while (suspendRunningThreads(count++)
-			       || startingThreadCount.load() > 0) {
+
+		while (suspendRunningThreads(count++)) {
 			import sys.posix.sched;
 			sched_yield();
 		}
 	}
 
 	void restartTheWorld() shared {
+		// Allow thread creation again.
+		createThreadLock.exclusiveUnlock();
```

Comment:
Why did you move this before sending the signals?

Reply:
Before I had allow thread creation before sending signals, and the locks were different.

Moving it down.